pages/test-pickup-times/index.tsx

```diff
@@ -0,0 +1,416 @@
+    requestTime: todayNMinutesFromNow(-30),
+  },
+  "SASB EDD": { holdingLocation: "mab88", fulfillment: "fulfillment:sasb-edd" },
+  "ReCAP EDD": {
+    holdingLocation: "rc123",
+    fulfillment: "fulfillment:recap-edd",
+  },
+  "ReCAP EDD, requested 3 days ago": {
+    holdingLocation: "rc123",
+    fulfillment: "fulfillment:recap-edd",
+    requestTime: todayNMinutesFromNow(-60 * 24 * 3),
+  },
+}
+const scenarioOptions = [""]
+  .concat(Object.keys(scenarios))
+  .reduce((a, label) => a.concat([{ label, value: label }]), [])
+
+const validDate = (dateString) => {
+  let time = null
+  try {
+    time = new Date(dateString).getTime()
+  } catch (e) {
+    console.error(`Invalid date: ${dateString}`)
+  }
+
+  return !!time
+}
+
+interface CurrentParams {
+  fulfillment: string
+  holdingLocation: string
+  deliveryLocation: string
+  requestTime: string
+  currentTime: string
+  scenario: string
+}
+
+/**
+ * The Search page is responsible for fetching and displaying the Search results,
+ * as well as displaying and controlling pagination and search filters.
+ */
+const TestPickupTimes = (params) => {
+  const router = useRouter()
+  const inputRef = useRef<TextInputRefType>()
+
+  const [holdingLocation, setHoldingLocation] = useState(
+    params.holdingLocation || "mal92"
+  )
+  const [fulfillment, setFulfillment] = useState(
+    params.fulfillment || fulfillmentOptions[0].value
+  )
+  const [deliveryLocation, setDeliveryLocation] = useState(
+    params.deliveryLocation || ""
+  )
+
+  const result = params.result
+
+  const [requestTime, setRequestTime] = useState(params.requestTime || "")
+  const [currentTime, setCurrentTime] = useState(params.currentTime || "")
+
+  const [currentParams, setCurrentParams] = useState({} as CurrentParams)
+  const error = params.error
+  const [loading, setLoading] = useState(false)
+
+  const [scenario, setScenario] = useState("")
+
+  const updateRoute = () => {
+    const queryString = stringify(currentParams)
+    router.push(`/test-pickup-times?${queryString}`)
+  }
+
+  useEffect(updateRoute, [currentParams])
+
+  if (currentParams && scenario !== currentParams.scenario) {
+    const scenarioParams = scenarios[scenario]
+    if (scenarioParams) {
+      setHoldingLocation(scenarioParams.holdingLocation || "")
+      setFulfillment(scenarioParams.fulfillment || "")
+      setDeliveryLocation(scenarioParams.deliveryLocation || "")
+      setRequestTime(scenarioParams.requestTime || "")
+      setCurrentTime(scenarioParams.currentTime || "")
+    }
+  }
+
+  if (
+    currentParams.fulfillment !== fulfillment ||
+    currentParams.holdingLocation !== holdingLocation ||
+    currentParams.deliveryLocation !== deliveryLocation ||
+    currentParams.requestTime !== requestTime ||
+    currentParams.currentTime !== currentTime
+  ) {
+    if (scenario && scenarios[scenario]) {
+      Object.keys(scenarios[scenario]).forEach((key) => {
+        if (scenarios[scenario][key] !== eval(key)) {
```

Comment:
This seems like a lot of nested if statements, would it be possible to refactor this a bit?

Reply:
Cleaned this up